### ISSUE TITLE
Certain tether start failures may not propagate to port layer

### ISSUE THREAD

**Issue by dougm** [vmware/vic#2327]
Container.start powers on the vm and waits for the `$id.started` key to be set in the vm extra config.  It is expected the value of this key will be "true" for success, anything else is considered an error:
https://github.com/vmware/vic/blob/0f6cb8929a99500f9ce634450480a47d417993c7/lib/portlayer/exec/container.go#L241-L244

`WaitForKeyInExtraConfig` ignores updates where the value of this key is set to "" or "&lt;nil&gt;".  If the tether fails to start without setting this value, the vm powers off and portlayer continues to wait.  In the past this surfaced as "Context Deadline Exceeded", but as of #2315 removing the timeout, will block forever.

The specific error seen in the tether debug log:

```
failed to apply network endpoint config: unable to acquire reference to link 192: more than one eth interface matches /sys/bus/pci/devices/0000:00:16.0/0000:*:00.0/net/*"
```

There are multiple issues here:

* We need to use a timeout in WaitForKeyInExtraConfig, just in case.

* Tether needs to set the SessionConfig.Started field in case(s) such as this

* The specific PCI path mapping error above

* Reduce mutex locking in the port layer; without the timeout, the mutex continues to hold, making it impossible to docker rm the container as it will block waiting to acquire the mutex

**Comment by dougm** [comment]
Timeout is covered by #2332 

Let's focus next on fixing the cases where the tether SessionConfig.Started field isn't being set.

We can create separate issues for the other 2 items later.

**Comment by dougm** [comment]
Regarding "more than one eth interface matches", the message is misleading.. the problem is that there are *zero* matches.  Short term fix would be to retry as we do for disk device PCI paths:

https://github.com/vmware/vic/blob/fde9102c937986487e8735e79a34f9a1a12abee5/pkg/vsphere/disk/disk_manager.go#L108

**Comment by mhagen-vmware** [comment]
Note: This error results in a failure to start a container. To workaround, wait until the system is less loaded and try again.

**Comment by dougm** [comment]
Retry checking for the ethernet device PCI path did not help.  Looking at the output of `lsmod` (via tether init), the vmxnet3 driver is *not* loaded when this happens.  Adding a call to `modprobe vmxnet3` in the tether seems to avoid the problem.

**Comment by dougm** [comment]
Root of this issue is fixed in #2407 - where the fix is to explicitly load the vmxnet3 driver.

**Comment by stuclem** [comment]
This is fixed. Removing kind/note.
